### PR TITLE
Add self reference test case for DoctrineTargetEntityStringToClassConstantRector

### PR DESCRIPTION
**Pull request by TomasVotruba**
Test fixture for https://github.com/rectorphp/rector/issues/8081

### CODE REVIEW

File: rules-tests/CodeQuality/Rector/Property/DoctrineTargetEntityStringToClassConstantRector/Fixture/Attribute/keep_self_reference.php.inc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Rector\Doctrine\Tests\CodeQuality\Rector\Property\DoctrineTargetEntityStringToClassConstantRector\Fixture\Attribute;
+
+use Doctrine\ORM\Mapping as ORM;
+
+final class KeepSelfReference
```
Comment (OskarStark) [inline rules-tests/CodeQuality/Rector/Property/DoctrineTargetEntityStringToClassConstantRector/Fixture/Attribute/keep_self_reference.php.inc:7]: Do we need the Entity Attribute here?

Comment (TomasVotruba) [reply rules-tests/CodeQuality/Rector/Property/DoctrineTargetEntityStringToClassConstantRector/Fixture/Attribute/keep_self_reference.php.inc:8]: Good point, just added :+1:

Comment (TomasVotruba) [reply rules-tests/CodeQuality/Rector/Property/DoctrineTargetEntityStringToClassConstantRector/Fixture/Attribute/keep_self_reference.php.inc:8]: Checking other fixtures, it's actually not needed. This rules checks directly these annotations/attributes.

There already is another test fixture for `self::class` as well :+1:

Comment (OskarStark) [reply rules-tests/CodeQuality/Rector/Property/DoctrineTargetEntityStringToClassConstantRector/Fixture/Attribute/keep_self_reference.php.inc:8]: Maybe add:
`#[ORM\JoinColumn(name: 'some_entity', onDelete: 'SET NULL')`

from my screenshot

Comment (OskarStark) [reply rules-tests/CodeQuality/Rector/Property/DoctrineTargetEntityStringToClassConstantRector/Fixture/Attribute/keep_self_reference.php.inc:8]: Ah the difference is, in your existing test you use `private ?self.....`

while here is no type hint.

Please do:
```diff
- private $someEntity;
+ private KeepSelfReference $someEntity;
```

if this does not work, try to make it nullable:

```diff
- private $someEntity;
+ private ?KeepSelfReference $someEntity;
```

**Comment by samsonasik** [comment]
@TomasVotruba @OskarStark it already resolved in PR:

- https://github.com/rectorphp/rector-doctrine/pull/208

please use rector dev-main

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Pull request merged**